### PR TITLE
#18409 Use FQN of the type when altering column's type

### LINKED ISSUE #18409: Postgres enum type editing inconvience
### Description

dbeaver_enums

### DBeaver Version

22.2.5.202211202223 CE

### Operating System

win

### Database and driver

PG

### Steps to reproduce

lets see at the picture. I have two schemas - repl and repl_log. I have some enums (custom types) declared in both of them.

There are bugs:
1. When I change type in the table editing dialog - i see only type names (without schemas, even when table schema differs from type schema);
2. When I try to save table - it fails, generating SQL:

ALTER TABLE [X] ALTER COLUMN [Y] TYPE [type] USING [Y]::[type];
but it should be:
ALTER TABLE [X] ALTER COLUMN [Y] TYPE [schema.type] USING [Y]::text::[schema.type];

Nota bene: enums cannot be converted directly (by 'USING'), but can be converted via additional 'text'

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ShadelessFox**
Acceptance criteria:
- [x] Column's type can be changed to a type from the same schema
- [x] Column's type can be changed to a type from another schema

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java [hunk 1]
```diff
@@ -207,9 +207,15 @@ protected void addObjectModifyActions(DBRProgressMonitor monitor, DBCExecutionCo
 //        ALTER [ COLUMN ] column SET STORAGE { PLAIN | EXTERNAL | EXTENDED | MAIN }
         String prefix = "ALTER " + table.getTableTypeName() + " " + DBUtils.getObjectFullName(table, DBPEvaluationContext.DDL) +
             " ALTER COLUMN " + DBUtils.getQuotedIdentifier(column) + " ";
-        String typeClause = column.getFullTypeName();
-        if (column.getDataSource().getServerType().supportsAlterTableColumnWithUSING() && column.getDataType() != null) {
-            typeClause += " USING " + DBUtils.getQuotedIdentifier(column) + "::" + column.getDataType().getName();
+        final PostgreDataType type = column.getDataType();
+        final String fullTypeName = type != null ? DBUtils.getObjectFullName(type, DBPEvaluationContext.DDL) : column.getFullTypeName();
+        String typeClause = fullTypeName;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java:212]: I think it is better to use existing PostgreDataType.getConditionTypeCasting or PostgreDialect.getTypeCastClause

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java [hunk 2]
```diff
@@ -207,9 +207,13 @@ protected void addObjectModifyActions(DBRProgressMonitor monitor, DBCExecutionCo
...
+        if (column.getDataSource().getServerType().supportsAlterTableColumnWithUSING()) {
+            typeClause += " USING ";
+            typeClause += column.getDataSource().getSQLDialect().getTypeCastClause(column, DBUtils.getQuotedIdentifier(column), true);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java:215]: getTypeCastClause already appends `::type-fqn`

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java (hunk 2 above)
Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/edit/PostgreTableColumnManager.java:215]: It doesn't. We need a sausage in a form of `[column_name]::text::[target_type]` for enums and `[column_name]::[target_type]` for other types.

The `getTypeCastClause` appends just the `::text` part for enums and several other types, but we still need to cast to the target type.

**Review by E1izabeth — APPROVED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**